test_apps/smokeTests/gridTools.cpp

```diff
@@ -0,0 +1,594 @@
+}
+
+void MakeRampOnAxis( Grid* grid, float minVal, float maxVal, size_t axis=X ) {
+    std::vector< size_t > dims = grid->GetDimensions();
+    size_t x = dims[X];
+    size_t y = dims[Y];
+    size_t z = 1;
+    if ( dims.size() == 3 ) 
+        z = dims[Z];
+
+    float xIncrement = axis==X ? (maxVal - minVal) / (dims[X]-1) : 0;
+    float yIncrement = axis==Y ? (maxVal - minVal) / (dims[Y]-1) : 0;
+    float zIncrement = axis==Z ? (maxVal - minVal) / (dims[Z]-1) : 0;
+
+    float value = minVal;
+    for ( size_t k=0; k<z; k++ ) {
+        value = axis==Y ? minVal : value;      // reset value if we're ramping on Y
+        for ( size_t j=0; j<y; j++ ) {
+            value = axis==X ? minVal : value;  // reset value if we're ramping on X
+            for ( size_t i=0; i<x; i++ ) {
+                grid->SetValueIJK( i, j, k, value );
+                value += xIncrement;
+            }
+            value += yIncrement;
+        }
+        value += zIncrement;
+    }
+}
+
+// This function iterates across all nodes in a grid; making comparisons between the
+// data values returned by the functions GetValueAtIndex() and GetValue().
+bool CompareIndexToCoords( 
+    VAPoR::Grid* grid, 
+    double &rms,                // Root Mean Square error
+    size_t &numMissingValues,   // Counter for receiving MissingValue upon query
+    size_t &disagreements      // Counter for when AccessIJK() and GetValue() disagree
+) {
+    rms   = 0.f;
+    disagreements  = 0;
+    numMissingValues = 0;
+
+    std::vector< size_t > dims = grid->GetDimensions();
+    size_t x = dims[X];
+    size_t y = dims[Y];
+    size_t z = 1;
+    if ( dims.size() == 3 ) 
+        z = dims[Z];
+ 
+    double peak  = 0.f;
+    double sum     = 0;
+    for ( size_t k=0; k<z; k++ ) {
+        for ( size_t j=0; j<y; j++ ) {
+            for ( size_t i=0; i<x; i++ ) {
+                size_t indices[3] = { i, j, k };
+                //float trueValue = grid.AccessIJK( i, j, k );
+                double trueValue = grid->GetValueAtIndex( {i,j,k} );
+                
+                double coords[3];
+                grid->GetUserCoordinates( indices, coords );
+                float sampleValue = grid->GetValue( coords );
+                
+                if (sampleValue == grid->GetMissingValue()) {
+                    numMissingValues++;
+                    continue;
+                }
+
+                double error = abs( sampleValue - trueValue );
+
+                if (error != 0) {
+                    disagreements++;
+                }
+
+                if ( error > peak ) 
+                    peak = error;
+                sum += error*error;
+
+            }
+        }
+    }
+
+    rms = sqrt( sum / (x*y*z) );
+
+    if ( rms != 0 || disagreements > 0 )
+        return 1;
+    return 0;
+}
+
+bool isNotEqual(
+    double x, 
+    double y
+) {
+  const double epsilon = 1e-5;
+  return std::abs(x - y) > epsilon * std::abs(x);
+}
+
+size_t TestConstNodeIterator( 
```

Comment:
Per earlier comments having the test function return a boolean to indicate success or failure (rather than an  would make the status of the test much more clear. It's really hard to understand under what conditions this test fails by reading the code; it returns expectedCount, that has nothing to do with what is being tested.

Reply:
Fixed.